ELIPs/ELIP-006.md

```diff
@@ -241,403 +238,25 @@ sequenceDiagram
-    participant SEF as Slash Escrow Factory
     participant STR as Strategy Contract
-    participant CL as Slash Escrow Clone
     participant RR as Redistribution Recipient
 
     Note over AVS,RR: Slashing Initiation
     AVS->>ALM: slashOperator<br>(avs, slashParams)
     ALM-->>DM: *Internal* <br>slashOperatorShares<br>(operator, strategies,<br> prevMaxMags, newMaxMags)
     Note over DM,SM: Share Management
     DM-->>SM: *Internal*<br>increaseBurnOrRedistributableShares<br>(operatorSet, slashId, strategy, addedSharesToBurn)
-    Note over SM,SEF: Escrow Setup (Initiates 4-day escrow period)
-    SM-->>SEF: *Internal* initiateSlashEscrow()
-    SEF-->>CL: *Internal* Deploy child proxy
-    
-    Note over SM,CL: Transfers underlying tokens to the `SlashEscrow` clone (Second Transaction)
-    SM->>CL: clearBurnOrRedistributableShares(operatorSet, slashId)
-    SM-->>STR: *Internal*<br>withdraw<br>(slashEscrow, token, underlyingAmount)
     
-    Note over SEF,SEF: Wait for max(globalDelay, strategyDelay) to elapse.
-    SEF->>SEF: getStrategyEscrowDelay()
-    
-    Note over SEF,CL: Final Distribution
-    RR ->>SEF: releaseSlashEscrow()
-    SEF-->>CL: *Internal* clearBurnOrRedistributableShares()
-    CL-->>RR: *Internal* <br>releaseTokens()
+    Note over SM,RR: Direct Fund Distribution
+    SM->>SM: clearBurnOrRedistributableShares(operatorSet, slashId)
+    SM-->>STR: *Internal*<br>withdraw<br>(recipient, token, underlyingAmount)
+    STR-->>RR: *Internal*<br>transfer<br>(token, underlyingAmount)
     Note right of RR: Final protocol fund outflow
 ```
 
-The rationale for this new contract, process, and delay is [outlined in the rationale](./ELIP-006.md#outflow-delay). A global minimum escrow period is introduced that is set by governance. This is a constant value, set at a minimum of four days. Strategies (staked assets) can have larger delays as well; this includes EIGEN, which will use a larger (14 day) delay to accommodate upcoming security features. The delay can be set via governance; Strategy confuguration is reserved for future protocol compatability and need.
-
-Previously, funds would be slashed and exited in a single step, with funds being marked to burn and a continuously running cron job executing fund exits. This was done non-atomically with slashing to maintain the guarantee that a slash should never fail, in the case where a token transfer or some other upstream issue of removing funds from the protocol may fail.
+Previously, funds would be slashed and exited in a single step, with funds being marked to burn and a continuously running cron job executing fund exits. This was done non-atomically with slashing to maintain the guarantee that a slash should never fail, in the case where a token transfer or some other upstream issue of removing funds from the protocol may fail. The new flow maintains this non-atomic approach while enabling direct distribution to redistribution recipients without a delay.
```

Comment:
> Previously, funds would be slashed and exited in a single step

I think this is a little confusing because you're combining what our offchain system was doing with the contract-level behavior.

The existing flow:
- marks shares slashed as soon as they're slashed
- a second, permissionless function is called to actually convert the shares to tokens and then burn them

We had a cron job calling the second function. The only change with redistribution is that the second function sends to the recipient, if applicable. (and I guess we're not running a cron job?)